realm/realm-annotations-processor/src/test/resources/io/realm/SimpleRealmProxy.java

```diff
@@ -263,7 +263,7 @@ public static String getTableName() {
     }
 
     public static some.test.Simple copyOrUpdate(Realm realm, some.test.Simple object, boolean update, Map<RealmModel,RealmObjectProxy> cache) {
-        if (object instanceof RealmObjectProxy && ((RealmObjectProxy) object).realmGet$proxyState().getRealm$realm() != null && ((RealmObjectProxy) object).realmGet$proxyState().getRealm$realm().threadId != realm.threadId) {
+        if (object instanceof RealmObjectProxy && ((RealmObjectProxy) object).realmGet$proxyState().getRealm$realm() != null && ((RealmObjectProxy) object).realmGet$proxyState().getRealm$realm().creatorThread != realm.creatorThread) {
```

Comment:
I cannot think of any reason that this won't work (... and I *did* try!).  None the less, it is just weird enough (comparing objects with == instead of .equals) that I would be inclined to put it into a small method, and comment the method.  Better yet, unless there is a *really* good reason for doing it, let's just use .equals.